homeassistant/components/motionblinds_ble/config_flow.py

```diff
@@ -0,0 +1,222 @@
+
+            assert self._discovery_info is not None
+
+            return self.async_create_entry(
+                title=str(self._display_name),
+                data={
+                    CONF_ADDRESS: self._discovery_info.address,
+                    CONF_LOCAL_NAME: self._discovery_info.name,
+                    CONF_MAC_CODE: self._mac_code,
+                    CONF_BLIND_TYPE: self._blind_type,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_BLIND_TYPE): SelectSelector(
+                        SelectSelectorConfig(
+                            options=[
+                                blind_type.value for blind_type in MotionBlindType
+                            ],
+                            translation_key=CONF_BLIND_TYPE,
+                            mode=SelectSelectorMode.DROPDOWN,
+                        )
+                    )
+                }
+            ),
+            description_placeholders={"display_name": self._display_name},
+        )
+
+    async def async_discover_motionblind(self, mac_code: str) -> None:
+        """Discover MotionBlinds initialized by the user."""
+        if not is_valid_mac(mac_code):
+            _LOGGER.error("Invalid MAC code: %s", mac_code.upper())
+            raise InvalidMACCode()
+
+        count = bluetooth.async_scanner_count(self.hass, connectable=True)
+        if count == 0:
+            self.hass.async_create_task(
+                self.hass.config_entries.flow.async_configure(flow_id=self.flow_id)
+            )
+            _LOGGER.error("No bluetooth adapter found")
+            raise NoBluetoothAdapter()
+
+        bleak_scanner = bluetooth.async_get_scanner(self.hass)
+        devices = await bleak_scanner.discover()
+
+        if len(devices) == 0:
+            _LOGGER.error("Could not find any bluetooth devices")
+            raise NoDevicesFound()
+
+        motion_device: BLEDevice | None = next(
+            (
+                device
+                for device in devices
+                if device
+                and device.name
+                and f"MOTION_{mac_code.upper()}" in device.name
+            ),
+            None,
+        )
+
+        existing_entries = self._async_current_entries()
+
+        if not motion_device:
+            _LOGGER.error("Could not find a motor with MAC code: %s", mac_code.upper())
+            raise CouldNotFindMotor()
+
+        unique_id = motion_device.address
+        if any(entry.unique_id == unique_id for entry in existing_entries):
+            _LOGGER.error(
+                "Device with MAC code %s has already been configured", mac_code.upper()
+            )
+            raise AlreadyConfigured()
+        await self.async_set_unique_id(unique_id, raise_on_progress=False)
+        self._discovery_info = motion_device
+        self._mac_code = mac_code.upper()
```

Comment:
Maybe it's better to use `homeassistant.helpers.device_registry.format_mac` to enforce a common format for the mac address?

Reply:
Disregard, I didn't realize `mac_code` is just 4 characters and not the complete mac address.